python/cudf/cudf/core/indexing.py

```diff
@@ -266,11 +266,35 @@ def _getitem_tuple_arg(self, arg):
             else:
                 return columns_df.index._get_row_major(columns_df, arg[0])
         else:
-            df = DataFrame()
-            for col in columns_df.columns:
-                # need Series() in case a scalar is returned
-                df[col] = Series(columns_df[col].loc[arg[0]])
-            df.columns = columns_df.columns
+            if isinstance(arg[0], slice):
+                start_index, stop_index = columns_df.index.find_label_range(
+                    arg[0].start, arg[0].stop
+                )
+
+                pos_slice = slice(start_index, stop_index, arg[0].step)
+                df = columns_df[pos_slice]
+            elif is_scalar(arg[0]):
+                try:
+                    index = column.as_column(
+                        columns_df.index
+                    ).find_first_value(arg[0], closest=False)
+                except (TypeError, KeyError, IndexError, ValueError):
+                    raise IndexError("label scalar is out of bound")
```

Comment:
Why do we need to catch on all of these different types of exceptions? I assume it should just be a specific exception from the `find_first_value` call failing?

Additionally, if we have a scalar against a duplicated index, shouldn't we return all of the values associated with the scalar?

Reply:
If that doesn't affect any downstream logic I would be in favor of just treating the scalar as another column.

Reply:
Using join for scalar as well with downstream logic.